synapse/storage/state.py

```diff
@@ -60,6 +61,13 @@ def __init__(self, hs):
             self.STATE_GROUP_INDEX_UPDATE_NAME,
             self._background_index_state,
         )
+        self.register_background_index_update(
+            self.CURRENT_STATE_INDEX_UPDATE_NAME,
+            index_name="current_state_events_member_index",
+            table="current_state_events",
+            columns=["state_key"],
```

Comment:
And the users_in_room is already handled by the `(room_id, type, state_key)` index.